plutus-contract/test/Spec/TxConstraints/MustMint.hs

```diff
@@ -94,6 +106,47 @@ mustMintCurrencyContract = do
     ledgerTx1 <- submitTxConstraintsWith @UnitTest lookups1 tx1
     awaitTxConfirmed $ Tx.getCardanoTxId ledgerTx1
 
+{-# INLINABLE mkMustReferenceOutputV2Validator #-}
+mkMustReferenceOutputV2Validator :: TxOutRef -> () -> PV2.ScriptContext -> Bool
+mkMustReferenceOutputV2Validator txOutRef _ =
+    TCV2.checkScriptContext @Void @Void (TC.mustReferenceOutput txOutRef)
+
+{-# INLINABLE mustReferenceOutputV2Validator #-}
+mustReferenceOutputV2Validator :: PV2.Validator
+mustReferenceOutputV2Validator = PV2.mkValidatorScript
+    $$(PlutusTx.compile [|| wrap ||])
+ where
+     wrap = PV2.mkUntypedValidator mkMustReferenceOutputV2Validator
+
+mustReferenceOutputV2ValidatorAddress :: Address
+mustReferenceOutputV2ValidatorAddress =
+    PV2.mkValidatorAddress mustReferenceOutputV2Validator
+
+mustMintValueWithReferenceContract :: Contract () Empty ContractError ()
+mustMintValueWithReferenceContract = do
+    utxos <- ownUtxos
+    myAddr <- Con.ownAddress
+    let (utxoRef, utxo) = head $ drop 5 $ Map.toList utxos
+        MintingPolicyHash mph = coinMintingPolicyHashV2
+        lookups0 = Constraints.plutusV2MintingPolicy coinMintingPolicyV2
```

Comment:
Should also use a minting policy with one of the new constraints to prove the phase-2 check passes.